src/extension/flutter/flutter_daemon.ts

```diff
@@ -72,6 +73,18 @@ export class FlutterDaemon extends StdIOService<UnknownNotification> implements
 		}
 	}
 
+	public startPing() {
+		const intervalId = setInterval(async () =>  {
+			try {
+				const result = await withTimeout(this.daemonVersion(), "The daemon connection was lost", 1);
```

Comment:
I was a bit concerned by this timeout, but it turns out there are two `withTimeout`s and one of them is in `ms` and one in `s` 😄 (my fault, but the one in ms is just in the legacy debug adapter and not used anywhere else).

Although, perhaps every second is still a bit frequent if it's going over a network. Would doing it every 10s be a much worse experience?

(Good idea to use `version` to avoid needing a new protocol message :-) )

Reply:
Ah sorry yeah, the timeout numbers are only for testing. 10s sounds fine to me for this check, and I could do a much larger number for how often this pings, like every minute or something.